FIX: UI stops working when multiple scenes with event systems are additively loaded and unloaded

andrew-oc opened the pull request:
### Description

Load a scene that contains an InputSystemUIInputModule that uses the default actions setup. Then additively load a second scene that also has an InputSystemUIInputModule with the default setup and unload the first scene. The first InputSystemUIInputModule will be destroyed and disable all of the input actions that the second input module is still using.

### Changes made

Added an InputAction reference counting solution to InputSystemUIInputModule where any time an InputSystemUIInputModule enables an action, it increments a reference for that action. When an InputSystemUIInputModule is destroyed, it will only disable input actions that have no more references. 

### Notes

The InputSystemUIInputModule used to have the concept of ownership of input actions to try and prevent this problem but it would fail for the case where the module was destroyed after another one had been created because ownership couldn't float between them. This might have been a reasonable fix for the issue here, but it would not have handled the case where an input action of the input module was assigned to after the module was enabled.

### Checklist

Before review:

- [x] Changelog entry added.
- [x] Tests added

andrew-oc committed:
FIX: UI stops working when scenes with event systems are additively loaded

andrew-oc committed:
Fix UnityEditor reference

andrew-oc committed:
Fix changelog merge conflict

Rene-Damm reviewed (approved):
LGTM.

Guess the problem here is the collapsing of configuration and runtime data into one showing its ugly side.

Wonder if there's a non-breaking way we can move away from that. Something where we have a controller object that for now is a step towards separation but remains optional and in addition to what's already there. And then later, when we can break, we move over entirely. Downside is it adds even more to an already crowded API surface.

andrew-oc committed:
Added prebuild setup steps for tests

Some InputSystemUIInputModule tests need a custom scene to be present in the build settings for test run in the built player
